### PR TITLE
Devas1705 BrightMirror create query

### PR DESCRIPTION
**Pull request by cedrtang**
DEVAS #1705:

- CreateBrightMirror mutation: create a brightMirror instance, reuse of table Idea, inspired from CreateIdea mutation and CreateThematic mutation. 

- Query BrightMirror: fetch all BrightMirror instances. Reuse of IdeaInterface. Inspired from Thematic query.

Unit tests added for empty and required cases. All global unit tests are passing

### CODE REVIEW

File: assembl/graphql/idea.py [hunk 1]
```diff
@@ -689,19 +584,26 @@ class Input:
...
 
     @staticmethod
     @abort_transaction_on_exception
     def mutate(root, args, context, info):
         EMBED_ATTACHMENT = models.AttachmentPurpose.EMBED_ATTACHMENT.value
         MEDIA_ATTACHMENT = models.AttachmentPurpose.MEDIA_ATTACHMENT.value
-        cls = models.Thematic
+        cls = models.Idea
+        phase_identifier = args.get('identifier')
+        if phase_identifier == 'survey':
```
Comment (amensouissi) [inline assembl/graphql/idea.py]: Please use 'Phases' Enum here (Phases.survey). See models.timeline (develop branche).

File: assembl/graphql/idea.py [hunk 2]
```diff
@@ -760,27 +663,59 @@ def mutate(root, args, context, info):
...
-            # `identifier` phase, child of the root idea,
-            # and add our thematic as a child of this root thematic.
-            root_thematic = get_root_thematic_for_phase(discussion, identifier)
-            if root_thematic is None:
-                root_thematic = create_root_thematic(discussion, identifier)
+            parent_idea_id = args.get('parent_id')
+            if parent_idea_id:
+                parent_idea_id = int(
+                    Node.from_global_id(parent_idea_id)[1])
+                if parent_idea_id:
```
Comment (amensouissi) [inline assembl/graphql/idea.py]: You don't need to test parent_idea_id here. int() or from_global_id raise an error if parent_idea_id is invalid .

File: assembl/graphql/idea.py [hunk 3]
```diff
@@ -760,27 +663,59 @@ def mutate(root, args, context, info):
...
+                        raise Exception('Parent Idea not found')
+                    if parent_idea.discussion != discussion:
+                        # No cross-debate references are allowed,
+                        # for security reasons
+                        raise Exception(
+                            'Parent Idea does not belong to this discussion')  # noqa: E501
+                else:
+                    raise Exception('Parent Idea not found')
+            else:
+                if phase_identifier in ('thread', 'multiColumns'):
```
Comment (amensouissi) [inline assembl/graphql/idea.py]: Use timeline.Phases to

File: assembl/graphql/idea.py [hunk 4]
```diff
@@ -108,7 +126,13 @@ def resolve_total_sentiments(self, args, context, info):
         return self.get_total_sentiments()
 
 
-class IdeaAnnoucement(SecureObjectType, SQLAlchemyObjectType):
+class IdeaAnnouncementInput(graphene.InputObjectType):
+    __doc__ = docs.IdeaAnnouncement.__doc__
+    title_entries = graphene.List(LangStringEntryInput, description=docs.IdeaAnnouncement.title_entries)
+    body_entries = graphene.List(LangStringEntryInput, description=docs.IdeaAnnouncement.body_entries)
+
```
Comment (amensouissi) [inline assembl/graphql/idea.py]: no required field here?

Comment (vincentfretin) [reply assembl/graphql/idea.py]: You're right, should be required, to have better flow definition to not check for null.

File: assembl/graphql/idea.py [hunk 5]
```diff
@@ -760,27 +663,59 @@ def mutate(root, args, context, info):
...
             db.add(
-                models.IdeaLink(source=root_thematic, target=saobj,
+                models.IdeaLink(source=parent_idea, target=saobj,
                                 order=args.get('order', order)))
 
+            # Create the idea announcement object which corresponds to the instructions
+            announcement = args.get('announcement')
+            if announcement is not None:
+                announcement_title_entries = announcement.get('title_entries')
+                if len(announcement_title_entries) == 0:
```
[the remark this answers is not in the document]
Comment (vincentfretin) [reply assembl/graphql/idea.py]: Required just means not null. Empty list is valid even if the field is required.

File: assembl/graphql/idea.py (hunk 3 above)
Comment (vincentfretin) [reply assembl/graphql/idea.py]: yes

File: assembl/graphql/idea.py (hunk 1 above)
Comment (vincentfretin) [reply assembl/graphql/idea.py]: yes

File: assembl/graphql/idea.py (hunk 2 above)
Comment (vincentfretin) [reply assembl/graphql/idea.py]: yep

**Pull request merged**